packages/components/badge/src/BadgeItem.tsx

```diff
@@ -0,0 +1,79 @@
+import { forwardRef, HTMLAttributes, PropsWithRef } from 'react'
+
+import { styles, type StylesProps } from './BadgeItem.styles'
+import {
+  DEFAULT_INTENT,
+  DEFAULT_LABELS,
+  DEFAULT_OVERFLOW_COUNT,
+  DEFAULT_SIZE,
+  DEFAULT_TYPE,
+} from './config'
+
+interface AccessibilityLabel {
+  noCount?: string
+  singleCount?: string
+  pluralCount?: string
+  moreThanOverflowCount?: string
+}
+
+export interface BadgeItemProps extends PropsWithRef<HTMLAttributes<HTMLSpanElement>>, StylesProps {
+  /**
+   * Numeric value used as indicator inside the component.
+   */
+  count?: number
+  /**
+   * Maximum numeric value to be dispayed as a count value.
+   */
+  overflowCount?: number
+  /**
+   * Object containing custom labels for accessibility purposes than can be set for each count case.
+   */
+  label?: AccessibilityLabel
```

Comment:
`label` as it is implemented seems too complicated for a DS.

1. `label` implies a string to me, not an object.
2. `label` implies it is an html label, not an aria-label.
3. Most DS keep things simple by simply opening an `aria-label` prop (string), the logic is then managed by them when they implement the component.